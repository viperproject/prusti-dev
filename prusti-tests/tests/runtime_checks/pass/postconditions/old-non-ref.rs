//@run
//@compile-flags: -Pcheck_overflows=false
use prusti_contracts::*;

#[derive(Clone)]
struct Something {
    pub field: i32,
}

// even though x.field is modified, since it not a reference
// if it occurrs in specifications it will be evaluated in
// its old state anyways
#[insert_runtime_check]
#[ensures(old(x.field) == x.field)]
fn nonsense(mut x: Something) {
    x.field = x.field + 4;
}

fn main() {
    let s = Something {
        field: 1,
    };
    nonsense(s);
}
