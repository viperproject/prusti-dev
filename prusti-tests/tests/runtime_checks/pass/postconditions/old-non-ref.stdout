check function prusti_post_check_item_nonsense is performed
